users/factories.py

```diff
@@ -14,6 +22,7 @@ class Meta:
     first_name = factory.Faker("first_name")
     last_name = factory.Faker("last_name")
     is_active = True
+    preferences = factory.RelatedFactory(UserPreferencesFactory, factory_related_name="user")
```

Comment:
I'm not sure why why you used here `RelatedFactory` instead of `SubFactory`. Can you explain?

Reply:
As we have a **user** (OneToOneField) in the `CustomUserPreferences` model we need to create this object for every user. Using `RelatedFactory` here we are able to do that. If we create a user by `UserFactory()` this automatically creates a `CustomUserPreferences` object for that **user**.
I based this implementation on [documentation](https://factoryboy.readthedocs.io/en/stable/recipes.html).